bokehjs/src/lib/models/annotations/scale_bar.ts

```diff
@@ -205,10 +216,21 @@ export class ScaleBarView extends AnnotationView {
     const frame_span = orientation == "horizontal" ? frame.bbox.width : frame.bbox.height
 
     const bar_length_percent = (() => {
-      if (0.0 <= bar_length && bar_length <= 1.0) {
-        return bar_length
-      } else {
-        return clamp(bar_length/frame_span, 0.0, 1.0)
+      const {bar_length, bar_length_units} = this.model
+      switch (bar_length_units) {
+        case "screen": {
+          if (0.0 <= bar_length && bar_length <= 1.0) {
+            return bar_length
+          } else {
+            return clamp(bar_length/frame_span, 0.0, 1.0)
+          }
+        }
+        case "data": {
+          const scale = orientation == "horizontal" ? this.coordinates.x_scale : this.coordinates.y_scale
+          const [sv0, sv1] = scale.r_compute(0, bar_length) // XXX assumes linear scale
```

Comment:
Let's note this assumption in the help string for the property or in some docs somewhere

Reply:
I added a note for this.